### DISCUSSION TITLE
How to mass export a folder of PSD files as flattened PNG images?

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [ImageMagick/ImageMagick#4709] [Help] [answered]
Hello. I am using Windows 10 X64

I would like to batch process a folder of PSD files and export them as flattened PNG images. So all layers are merged into one keeping transparent background

Original resolution and quality

How can I do that? Thank you so much for answers

**Comment by snibgo** [comment, marked as answer]
> ... export them as flattened PNG images. So all layers are merged into one keeping transparent background

I don't understand. "Flattening" implies removing transparency. What "transparent background" do you want to keep?

The first image in a PSD file is often a flattened version of the other images. Is that what you want?
```
magick in.psd[0] out.png
```

**Reply by FurkanGozukara** [reply]
> > ... export them as flattened PNG images. So all layers are merged into one keeping transparent background
> 
> I don't understand. "Flattening" implies removing transparency. What "transparent background" do you want to keep?
> 
> The first image in a PSD file is often a flattened version of the other images. Is that what you want?
> 
> ```
> magick in.psd[0] out.png
> ```

what i mean is merging layers with keeping transparent background

also I need help to batch process a folder

**Reply by fmw42** [reply]
If on IM 6, create a new directory e.g. test2. Then change directories to your image folder (e.g. test1). Then use mogrify. If on IM 7, use `magick mogrify` in place of `mogrify`

```
cd
cd desktop/test1
mogrify -path ../test2 -format png *.psd[0]

```

Results will be PNG in new folder (test2)

See https://legacy.imagemagick.org/Usage/basics/#mogrify

**Reply by FurkanGozukara** [reply]
@fmw42 this works awesome ty so much "mogrify -path ../test2 -format png *.psd[0]"

**Comment by snibgo** [comment]
To merge all the images in the PSD:
```
magick in.psd -layers merge out.png
```
To process all the files in a Windows directory, I suggest a "for" loop. Type "help for" for the syntax.
